3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,234 @@
+  console.log(travelTime); // Output: 8 hours and 38 minutes
+  
+  //Series duration of my life
+
+
+  const seriesDurations = [
+    {
+      title: "Game of Thrones",
+      days: 3,
+      hours: 1,
+      minutes: 0,
+    },
+    {
+      title: "Sopranos",
+      days: 3,
+      hours: 14,
+      minutes: 0,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+    // Add your favorite series here
+    {
+      title: "Dark",
+      days: 1,
+      hours: 10,
+      minutes: 0,
+    },
+  ];
+  
+  function calculateSeriesPercentage(seriesDurations) {
+    const averageLifespan = 80 * 365 * 24 * 60; // 80 years converted to minutes
+    let totalSeriesTime = 0;
+  
+    for (const series of seriesDurations) {
+      const seriesTimeMinutes =
+        series.days * 24 * 60 + series.hours * 60 + series.minutes;
+      totalSeriesTime += seriesTimeMinutes;
+    }
+  
+    const percentage = (totalSeriesTime / averageLifespan) * 100;
+  
+    return percentage.toFixed(3); // Format percentage to three decimal places
+  }
+  
+  function logOutSeriesText() {
+    for (const series of seriesDurations) {
+      const seriesPercentage = (
+        (series.days * 24 * 60 + series.hours * 60 + series.minutes) /
+        (80 * 365 * 24 * 60)
+      ).toFixed(3);
+      console.log(`${series.title} took ${seriesPercentage}% of my life`);
+    }
+  
+    const totalPercentage = calculateSeriesPercentage(seriesDurations);
+    console.log(`\nIn total, that is ${totalPercentage}% of my life`);
+  }
+  
+  logOutSeriesText();
+  
+//Save a note
+
+const notes = [];
+
+function saveNote(content, id) {
+  if (typeof id !== 'number') {
+    console.error('Error: ID must be a number.');
+    return;
+  }
+
+  const note = { content, id };
+  notes.push(note);
+}
+
+function getNote(id) {
+  if (typeof id !== 'number') {
+    console.error('Error: ID must be a number.');
+    return;
+  }
+
+  for (const note of notes) {
+    if (note.id === id) {
+      return note;
+    }
+  }
+
+  console.error(`Error: No note found with ID ${id}`);
+}
+
+function logOutNotesFormatted() {
+  for (const note of notes) {
+    console.log(`The note with id: ${note.id}, has the following note text: "${note.content}"`);
+  }
+}
+
+// Unique Feature: Add the ability to delete a note by its ID.
+function deleteNote(id) {
+  if (typeof id !== 'number') {
+    console.error('Error: ID must be a number.');
+    return;
+  }
+
+  const indexToDelete = notes.findIndex((note) => note.id === id);
+
+  if (indexToDelete !== -1) {
+    notes.splice(indexToDelete, 1);
+    console.log(`Note with ID ${id} has been deleted.`);
+  } else {
+    console.error(`Error: No note found with ID ${id}`);
+  }
+}
+
+// Example usage:
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes);
+
+const firstNote = getNote(1);
+console.log(firstNote);
+
+logOutNotesFormatted();
+
+deleteNote(2);
+console.log(notes);
+
```

Comment:
🎉 Firstly, great job! Your solution for the note-taking app is comprehensive and shows good coding practice, especially with the added type-checking for IDs.

👍 The usage of typeof to ensure the ID is a number is an excellent touch for data validation, ensuring the integrity of the data in the notes array.

✨ Suggestions:

- Consider implementing a function or mechanism to avoid the possibility of duplicate IDs. If two notes with the same ID are saved, the getNote and deleteNote functions might not behave as expected.

The `getNote` function can be enhanced to return an error or a message if the note with the specified ID is not found, similar to the `deleteNote` function.